contracts/timelocks/YakTimelockForDexStrategyV4.sol

```diff
@@ -0,0 +1,478 @@
+        require(msg.sender == feeCollector);
+        _;
+    }
+
+    /**
+     * @notice Set timelock when changing pending values
+     * @param _strategy address
+     * @param _fn Function enum value
+     * @param timelockLength in seconds
+     */
+    modifier setTimelock(address _strategy, Functions _fn, uint timelockLength) {
+        timelock[_strategy][_fn] = block.timestamp + timelockLength;
+        _;
+    }
+
+    /**
+     * @notice Enforce timelock for a given function
+     * @dev Ends execution by resetting timelock to avoid replay
+     * @param _strategy address
+     * @param _fn Function enum value
+     */
+    modifier enforceTimelock(address _strategy, Functions _fn) {
+        require(timelock[_strategy][_fn] != 0 && timelock[_strategy][_fn] <= block.timestamp, "YakTimelockManager::enforceTimelock");
+        _;
+        timelock[_strategy][_fn] = 0;
+    }
+
+    /**
+     * @notice Grant access for a time lock method for an account
+     * @param _ptf PermissionedTimelockFunctions enum value for the desired method
+     * @param account address to grant the access
+     */
+    function grantAccountAccess(PermissionedTimelockFunctions _ptf, address account) external onlyManager {
+        require(!authorizedAddressess[_ptf][account], "Account has already been given access");
+        emit grant(_ptf,account);
+        authorizedAddressess[_ptf][account] = true;
+        permissionedFunctionOwners[_ptf].push(account);
+    }
+
+    /**
+     * @notice Revokes access for a time lock method for an account
+     * @param _ptf PermissionedTimelockFunctions enum value for the desired method
+     * @param account address to revoke the access
+     */
+    function revokeAccountAccess(PermissionedTimelockFunctions _ptf, address account) external onlyManager {
+        require(authorizedAddressess[_ptf][account], "Account has no access");
+        emit revoke(_ptf,account);
+        uint accountIndex = permissionedFunctionOwners[_ptf].length+1;
+        for(uint i = 0; i < permissionedFunctionOwners[_ptf].length; i++) {
+            if (permissionedFunctionOwners[_ptf][i] == account) {
+                accountIndex = i;
+                break;
+            }
+        }
+        require(accountIndex < permissionedFunctionOwners[_ptf].length, "Account not found");
+        authorizedAddressess[_ptf][account] = false;
+        _removeElement(_ptf, accountIndex);
+    }
+
+    /**
+     * @notice returns a list of addressess that have grant access to a timelock function 
+     * @dev The timelock contract may receive assets from both revenue and asset recovery.
+     * @dev The sweep function is NOT timelocked, because recovered assets must go through separate timelock functions.
+     * @param _ptf PermissionedTimelockFunctions enum value for the desired method
+     */
+    function listofGrantedAccounts(PermissionedTimelockFunctions _ptf) external view onlyManager returns (address [] memory) {
```

Comment:
listOf

Reply:
I don't know if this works, generally, to navigate arrays in solidity, you need to give the front end a function with the array length, and then, another function, that with the _ptf and the index, it returns the address, and the front-end has to do the `for` on their end based on the length the other function provided and call one by one to get the addresses